conformance/lang/expressions/list_constructor.balt

```diff
@@ -0,0 +1,1374 @@
+Test-Case: parser-error
+Description: Test list constructor negative syntax.
+Labels: array-type, list-constructor-expr
+
+function errorFunction() {
+    int[] arr1 = [; // @error missing close bracket token
+    int[] arr2 = ]; // @error missing open bracket token
+    int[] arr3 = [
+        1
+        2 // @error missing comma token
+        3 // @error missing comma token
+    ];
+}
+
+Test-Case: output
+Description: Test basic list constructor expression without contextually expected type.
+Labels: list-constructor-expr, string
+
+function init() {
+    io:println([]); // @output []
```

Comment:
The description says "without a contextually expected type". But here, there is a contextually-expected type: the parameter type from `io:println()` - `io:Printable`.

No contextually-expected type is when used with `var`.